iib/workers/tasks/build.py

```diff
@@ -106,59 +106,70 @@ def _cleanup():
 
 
 @retry(attempts=3, wait_on=IIBError, logger=log)
-def _create_and_push_manifest_list(request_id, arches):
+def _create_and_push_manifest_list(request_id, arches, build_tags):
     """
     Create and push the manifest list to the configured registry.
 
     :param int request_id: the ID of the IIB build request
     :param iter arches: an iterable of arches to create the manifest list for
+    :param build_tags: list of extra tag to use for intermetdiate index image
     :return: the pull specification of the manifest list
     :rtype: str
     :raises IIBError: if creating or pushing the manifest list fails
     """
-    output_pull_spec = get_rebuilt_image_pull_spec(request_id)
-    log.info('Creating the manifest list %s', output_pull_spec)
-    with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
-        manifest_yaml = os.path.abspath(os.path.join(temp_dir, 'manifest.yaml'))
-        with open(manifest_yaml, 'w+') as manifest_yaml_f:
-            manifest_yaml_f.write(
-                textwrap.dedent(
-                    f'''\
-                    image: {output_pull_spec}
-                    manifests:
-                    '''
+    output_pull_specs = [get_rebuilt_image_pull_spec(request_id)]
+    if build_tags:
+        conf = get_worker_config()
+        for tag in build_tags:
+            output_pull_specs.append(
+                conf['iib_image_push_template'].format(
+                    registry=conf['iib_registry'], request_id=tag
                 )
             )
-            for arch in sorted(arches):
-                arch_pull_spec = _get_external_arch_pull_spec(request_id, arch)
-                log.debug(
-                    'Adding the manifest %s to the manifest list %s',
-                    arch_pull_spec,
-                    output_pull_spec,
-                )
+
+    for output_pull_spec in output_pull_specs:
```

Comment:
if I understand correctly, every pull spec will be appended with each arch?
Could you give me an example how the yaml file will look like in the end?

Reply:
that's correct. Yaml files will look very same as they were before this change

Reply:
thanks. I saw the example in tests